### PR TITLE
Add a List component for lists of things

### PR DESCRIPTION
**Pull request by kofi-clever**
**Jira:** https://clever.atlassian.net/browse/FEE-41
**Layout Guide:** https://docs.google.com/document/d/14Qs75OMztK4YzxlJdiqLZ5VbmFKQA5SDe99Ws5u87FU

**Overview:**
Add a `<List />` components for all your "1-column table" needs!

After consultation with Mischelle, setting defaults to bordered list with plain rows.

**Screenshots/GIFs:**
screen shot 2017-08-03 at 5 39 29 pm

https://gyazo.com/eabf23ab5db8037fe859a427b653dc2c

**Testing:**
- [N/A] Unit tests -- Component's a little too simple for tests. Demo's sufficient for now.
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [x] IE10

**Roll Out:**
- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change? Run `npm version minor`
    - Backward compatible change? Run `npm version patch`
    - Only changing documentation? All good. Skip this step.
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,7 +47,7 @@ border-radius-styles:
 	@echo "Generating border-radius style definitions..."
 	@node genBorderRadius.js
 
-LINT_MAX_LESS_PROBLEMS := 125
+LINT_MAX_LESS_PROBLEMS := 322
```
Comment (CarlNelson) [inline Makefile]: This seems like a lot of added problems...

bitmoji

File: src/List/List.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const RowType = {
+  PLAIN: "plain",
+  BORDERED: "bordered",
+  ZEBRA: "zebra",
+};
+
+const cssClass = {
+  CONTAINER: "List",
+  EMPTY: "List--empty",
+  HEADER: "List--header",
```
Comment (CarlNelson) [inline src/List/List.jsx]: This doesn't match our Dewey guidelines for components for naming. Only modifiers like noBorder or empty are camelCased, otherwise PascalCased.

File: src/Table/Table.jsx [hunk 3]
```diff
@@ -271,6 +271,13 @@ export class Table extends Component {
     const {lazy, numRows} = this.props;
     const {currentPage, sortState, pageLoading, allLoaded} = this.state;
 
+    const columns = lodash.compact(React.Children.toArray(children));
+    if (columns.length < 2) {
+      throw new Error(
+        "Table requires at least 2 columns. Consider using the List component instead."
```
Comment (CarlNelson) [inline src/Table/Table.jsx]: 💯

File: Makefile (hunk 1 above)
Comment (kofi-clever) [inline Makefile]: haha yeah...we had a broken lint rule that's been deprecated. upgraded to the current one and these popped up. they're mostly from the auto-generated files that don't have any new lines between rulesets, so we should be able to bring the count down a bit in a follow-up if we update the script

File: src/List/List.jsx (hunk 2 above)
Comment (kofi-clever) [inline src/List/List.jsx]: ah, the idea was to have PascalCase for top-level components (basically, match the javascript class name) - so far, we've been using camelCase for sub-elements and modifiers (i.e. `.List--Item` is a (sub-)component used with `List`, `.List--header` is a sub-element directly within `List`'s markup, `.List--header--small` would be a modified version of the header sub-element, etc.)

Comment (arsalansufi) [inline src/List/List.jsx]: TIL: Didn't realize we had this convention -- good to know!

File: docs/components/ListView.jsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+import Example, {CodeSample, ExampleCode} from "./Example";
+import PropDocumentation from "./PropDocumentation";
+import View from "./View";
+import {List, FlexBox, ItemAlign, SegmentedControl} from "src";
+
+import "./ListView.less";
+
+const Items = {
+  empty: [],
+  nonEmpty: [
```
Comment (peggyl) [inline docs/components/ListView.jsx]: 💯 This example is amazing.

File: docs/components/ListView.jsx [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        </Example>
+
+        <PropDocumentation
+          title="<List /> Props"
+          availableProps={[
+            {
+              name: "children",
+              type: "List.Item | List.Item[]",
+              description: "List items to display.",
+              optional: true,
```
Comment (peggyl) [inline docs/components/ListView.jsx]: Shouldn't this be required even if an empty list is accepted?

File: docs/components/ListView.jsx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+  state = {
+    itemsKey: "nonEmpty",
+    rowType: List.defaultProps.rowType,
+    showCustomEmptyMessage: false,
+    showTitle: true,
+    showBorder: true,
+  };
+
+  componentWillMount() {
```
Comment (peggyl) [inline docs/components/ListView.jsx]: Why is this needed? The entire component lifecycle is invoked even without an empty function, right?

File: src/List/List.jsx (hunk 2 above)
Comment (CarlNelson) [inline src/List/List.jsx]: We should clarify in the docs then, my assumption was that all elements were PascalCased and only modifiers were camelCased.

**Comment by CarlNelson** [comment]
LGTM

File: docs/components/ListView.jsx (hunk 6 above)
Comment (kofi-clever) [inline docs/components/ListView.jsx]: oops, good catch - i copied this file from another components docs and didn't fully cleanup

File: docs/components/ListView.jsx (hunk 5 above)
Comment (kofi-clever) [inline docs/components/ListView.jsx]: yeah, fair point. functionally, it's optional, but in practice, there's no reason for anyone to provide null/undefined children
